src/index.ts

```diff
@@ -1,10 +1,34 @@
-import { execSync } from 'node:child_process';
 import os from 'node:os';
+import path from 'node:path';
+import * as fs from 'node:fs';
+import { execSync } from 'node:child_process';
+import { fileURLToPath } from 'node:url';
 
 const WANTED_VERSION = 'v18.19.1';
 const MIN_NODE_UPDATE_MEMORY = 1e9;
+const dirname = path.dirname(fileURLToPath(import.meta.url));
+
+function updateEntrypoint () {
+	const currentEntrypointPath = path.join(dirname, '../../entrypoint.sh');
+	const newEntrypointPath = path.join(dirname, '../bin/entrypoint.sh');
+
+	if (!fs.existsSync(currentEntrypointPath) || !fs.existsSync(newEntrypointPath)) {
+		return;
+	}
+
+	const currentEntrypoint = fs.readFileSync(currentEntrypointPath, 'utf8');
+	const newEntrypoint = fs.readFileSync(newEntrypointPath, 'utf8');
+
+	if (currentEntrypoint === newEntrypoint) {
+		return;
+	}
+
+	console.log('Entrypoint change detected. Updating and restarting.');
+	fs.writeFileSync(currentEntrypointPath, `cp ${newEntrypointPath} ${currentEntrypointPath} && exit\n`);
```

Comment:
Is there a specific reason to `cp ...` instead of `fs.writeFileSync(newEntrypoint);`?

Reply:
Yes 😄 if you change a bash script while it's running, it continues reading from the new file at the original byte offset, which results in executing something random. But if the new file is shorter, like in this case, it simply ends.

Reply:
Maybe I'll add a comment about that in the code.